src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,482 @@
+import java.net.URISyntaxException;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.security.auth.x500.X500Principal;
+
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEFAULT_ENV_STAGE;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_MQTT_NAMESPACE;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_NETWORK_PROXY_NAMESPACE;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_AWS_REGION;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_ENV_STAGE;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_GG_DATA_ENDPOINT;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_GG_DATA_PLANE_PORT;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_IOT_DATA_ENDPOINT;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_NO_PROXY_ADDRESSES;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_PROXY_PASSWORD;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_PROXY_URL;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_PROXY_USERNAME;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PROXY_NAMESPACE;
+import static com.aws.greengrass.deployment.DeviceConfiguration.GG_DATA_PLANE_PORT_DEFAULT;
+import static com.aws.greengrass.deployment.DynamicComponentConfigurationValidator.DEFAULT_TIMEOUT_SECOND;
+import static com.aws.greengrass.mqttclient.MqttClient.DEFAULT_MQTT_KEEP_ALIVE_TIMEOUT;
+import static com.aws.greengrass.mqttclient.MqttClient.DEFAULT_MQTT_OPERATION_TIMEOUT;
+import static com.aws.greengrass.mqttclient.MqttClient.DEFAULT_MQTT_PING_TIMEOUT;
+import static com.aws.greengrass.mqttclient.MqttClient.DEFAULT_MQTT_PORT;
+import static com.aws.greengrass.mqttclient.MqttClient.DEFAULT_MQTT_SOCKET_TIMEOUT;
+import static com.aws.greengrass.mqttclient.MqttClient.MQTT_KEEP_ALIVE_TIMEOUT_KEY;
+import static com.aws.greengrass.mqttclient.MqttClient.MQTT_OPERATION_TIMEOUT_KEY;
+import static com.aws.greengrass.mqttclient.MqttClient.MQTT_PING_TIMEOUT_KEY;
+import static com.aws.greengrass.mqttclient.MqttClient.MQTT_PORT_KEY;
+import static com.aws.greengrass.mqttclient.MqttClient.MQTT_SOCKET_TIMEOUT_KEY;
+
+public class ConnectivityValidator implements AutoCloseable {
```

Comment:
I think that we do need to make this additional level of validation optional.

Reply:
customers can choose to skip the validation if their devices are configured to be offline or the validation timeout is set to 0